gtsfm/utils/verification.py

```diff
@@ -75,37 +80,48 @@ def essential_to_fundamental_matrix(
 
 
 def compute_epipolar_distances(
-    normalized_coords_i1: np.ndarray, normalized_coords_i2: np.ndarray, i2Ei1: EssentialMatrix
+    coordinates_i1: np.ndarray, coordinates_i2: np.ndarray, i2Fi1: np.ndarray, distance_type: str = "sed"
 ) -> Optional[np.ndarray]:
-    """Compute symmetric point-line epipolar distances between normalized coordinates of correspondences.
+    """Compute point-line epipolar distance between corresponding coordinates in two images.
 
     Args:
-        normalized_coords_i1: normalized coordinates in image i1, of shape Nx2.
-        normalized_coords_i2: corr. normalized coordinates in image i2, of shape Nx2.
-        i2Ei1: essential matrix between two images
+        coordinates_i1: coordinates in image i1, of shape Nx2.
+        coordinates_i2: corr. coordinates in image i2, of shape Nx2.
+        i2Fi1: fundamental matrix between two images.
+        distance_type (optional): type of distance metric to compute. The options are "sed" (symmetric epipolar
+                                  distance) and "sampson". Defaults to SED.
 
     Returns:
-        Symmetric epipolar distances for each row of the input, of shape N.
+        Epipolar point-line distances for each row of the input, of shape N.
     """
-    if (
-        normalized_coords_i1 is None
-        or normalized_coords_i1.size == 0
-        or normalized_coords_i2 is None
-        or normalized_coords_i2.size == 0
-    ):
+    if distance_type != "sed" and distance_type != "sampson":
+        raise ValueError("Invalid distance type for epipolar distances.")
+
+    if coordinates_i1 is None or coordinates_i1.size == 0 or coordinates_i2 is None or coordinates_i2.size == 0:
         return None
 
-    # construct the essential matrix in the opposite directin
-    i2Ti1 = Pose3(i2Ei1.rotation(), i2Ei1.direction().point3())
-    i1Ti2 = i2Ti1.inverse()
-    i1Ei2 = EssentialMatrix(i1Ti2.rotation(), Unit3(i1Ti2.translation()))
+    # # compute the pt line distance as done in Opencv
+    epipolar_lines_i2 = feature_utils.convert_to_epipolar_lines(coordinates_i1, i2Fi1)  # Ex1
+    epipolar_lines_i1 = feature_utils.convert_to_epipolar_lines(coordinates_i2, i2Fi1.T)  # Etx2
+
+    if distance_type == "sampson":
+
+        num = feature_utils.compute_point_line_distances(coordinates_i1, epipolar_lines_i1)
```

Comment:
should `def sampson()` just be a helper function? easier to unit test?

Reply:
`compute_epipolar_distances_sed` and `compute_epipolar_distances_sampson`?

Reply:
sure, sounds good

Reply:
resolved